athenzdomains.go

```diff
@@ -0,0 +1,183 @@
+			if err != nil {
+				fmt.Printf("Error occurred when converting assertion resource into regex format. Error: %v", err)
+			}
+			actionRegex, err := regexp.Compile("^" + replacer.Replace(strings.ToLower(assertion.Action)) + "$")
+			if err != nil {
+				fmt.Printf("Error occurred when converting assertion action into regex format. Error: %v", err)
+			}
+			simpleAssert := SimpleAssertion{
+				resource: resourceRegex,
+				action:   actionRegex,
+				effect:   effect,
+			}
+			crMap.RoleToAssertion[assertion.Role] = append(crMap.RoleToAssertion[assertion.Role], simpleAssert)
+		}
+
+	}
+}
+
+func (c *Cache) addObj(domainMap map[string]CRMap, item *v1.AthenzDomain) {
+	c.lock.Lock()
+	domainName := item.ObjectMeta.Name
+	_, ok := domainMap[domainName]
+	if !ok {
+		roleToPrincipals := make(map[string][]*zms.RoleMember)
+		roleToAssertion := make(map[string][]SimpleAssertion)
+		crMap := CRMap{
+			RoleToPrincipals: roleToPrincipals,
+			RoleToAssertion:  roleToAssertion,
+		}
+		domainMap[domainName] = crMap
+	}
+	parseData(domainMap, domainName, item)
+	c.lock.Unlock()
+}
+
+func (c *Cache) updateObj(domainMap map[string]CRMap, item *v1.AthenzDomain) {
+	c.lock.Lock()
+	domainName := item.ObjectMeta.Name
+	_, ok := domainMap[domainName]
+	if ok {
+		delete(domainMap, domainName)
+	}
+	roleToPrincipals := make(map[string][]*zms.RoleMember)
+	roleToAssertion := make(map[string][]SimpleAssertion)
+	crMap := CRMap{
+		RoleToPrincipals: roleToPrincipals,
+		RoleToAssertion:  roleToAssertion,
+	}
+	domainMap[domainName] = crMap
+	parseData(domainMap, domainName, item)
+	c.lock.Unlock()
+}
+
+func (c *Cache) deleteObj(domainMap map[string]CRMap, item *v1.AthenzDomain) {
+	c.lock.Lock()
+	domainName := item.ObjectMeta.Name
+	_, ok := domainMap[domainName]
+	if ok {
+		delete(domainMap, domainName)
+	}
+	c.lock.Unlock()
+}
+
+func (c *Cache) authorize(ctx context.Context, principal string, check AthenzAccessCheck) {
+	domainName := strings.Split(check.Resource, ":")
+	if len(domainName) != 2 {
+		fmt.Println("Error splitting domain name")
+	}
+
+	c.lock.RLock()
+	crMap := c.DomainMap
+	domainData := crMap[domainName[0]]
+	roles := []string{}
+	for roleName, member := range domainData.RoleToPrincipals {
+		for _, m := range member {
+			memberRegex, err := regexp.Compile("^" + replacer.Replace(strings.ToLower(string(m.MemberName))) + "$")
+			fmt.Println(memberRegex)
+			if err != nil {
+				fmt.Printf("Error occurred when converting memberNames in roleMember list into regex format. Error: %v", err)
+			}
+			if memberRegex.MatchString(principal) {
+				roles = append(roles, roleName)
+			}
+		}
+	}
+
+	for _, r := range roles {
+		policies := domainData.RoleToAssertion[r]
+		for _, assert := range policies {
+			if assert.resource.MatchString(check.Resource) && assert.action.MatchString(check.Action) && assert.effect == "ALLOW" {
+				fmt.Println("Authorization successful using cache data")
+			}
+		}
+	}
+	fmt.Println("Authorization failed")
```

Comment:
Can we remove all the prints in this function? We will print the results in the calling function.